src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -231,8 +230,8 @@ public Task Stop()
                             HttpContextFactory.Dispose(httpContext);
 
                             await ProduceEnd();
-
-                            while (await RequestBody.ReadAsync(_nullBuffer, 0, _nullBuffer.Length) != 0)
+                            
+                            while (await MessageBody.SkipAsync() != 0)
```

Comment:
I think this should be a single call instead of a loop.

`RequestBody.Consume()`